### PR TITLE
Remove redundant and unused microcodes + their supporting code (S2DEX2 and HVQM)

### PR DESCRIPTION
**Pull request by mountainflaw**
Removes every grucode except F3DEX and F3DZEX 2. 

- F3DZEX 2 (currently selectable as `f3dzex` already includes the point lighting functionality present in the PosLight microcodes, and is the current default of HackerSM64, making them redundant. 
- F3D and F3D_OLD served no benefit to keep around as they are obsoleted in every way by F3DEX, which will be kept around for its slightly more precise Z-buffer.
- Super3D was experimental and serves no purpose.
- n_aspMain is unusable with the current state of the audio library.
- HVQM is unusable.

This also removes the S2DEX2 and HVQM engines as they aren't really used by anyone and take up a lot of space.

XBUS microcode variants are also removed due to no practical benefit towards maintaining them.

### CODE REVIEW

**Comment by Reonu** [comment]
Based

**Comment by mountainflaw** [comment]
Ready!

**Review by arthurtilly — APPROVED** [review]

**Comment by mountainflaw** [comment]
Additional information:

- `NoN` (no nearfield clipping) is now enforced for all microcode options.
- `F3DEX2.Rej` is preserved as we do not have an `F3DZEX.Rej` to choose from.

**Comment by mountainflaw** [comment]
Rebased on 2.3.0.

File: rsp/audio.s [hunk 1]
```diff
@@ -1,1896 +0,0 @@
-.rsp
```
Comment (gheskett) [inline rsp/audio.s:1]: Why is this file being removed? There's no point other than to ensure no custom audio ucode ever gets written. Doesn't exactly count as bloat either considering the game actually uses this file.

**Review by gheskett — CHANGES_REQUESTED** [review]

File: rsp/audio.s (hunk 1 above)
Comment (mountainflaw) [reply rsp/audio.s:1]: `audio.s` was removed and replaced with `aspMain.o` to store the audio microcode in a way consistent with other microcodes.

**Comment by mountainflaw** [comment]
Succeeded by #816 .

**Pull request closed without merging**